### PR TITLE
docs(release): v1.29.1 release template improvements

### PR DESCRIPTION
**Pull request by BigLep**
This is for doing release process followups after executing https://github.com/filecoin-project/lotus/issues/12470

## Checklist

Before you mark the PR ready for review, please make sure that:

- [ ] Commits have a clear commit message.
- [ ] PR title conforms with [contribution conventions](https://github.com/filecoin-project/lotus/blob/master/CONTRIBUTING.md#pr-title-conventions)
- [ ] Update CHANGELOG.md or signal that this change does not need it per [contribution conventions](https://github.com/filecoin-project/lotus/blob/master/CONTRIBUTING.md#changelog-management)
- [ ] New features have usage guidelines and / or documentation updates in
  - [ ] [Lotus Documentation](https://lotus.filecoin.io)
  - [ ] [Discussion Tutorials](https://github.com/filecoin-project/lotus/discussions/categories/tutorials)
- [ ] Tests exist for new functionality or change in behavior
- [ ] CI is green

### CODE REVIEW

**Comment by BigLep** [comment]
This isn't ready for review yet. I'll either work on it today or 2024-09-17.  A couple of things:
1. It didn't have a step for copy/pasting in content to the GitHub release
2. There was a lot of cruft for when trying to rush out a release.

File: documentation/misc/RELEASE_ISSUE_TEMPLATE.md [hunk 1]
```diff
@@ -57,21 +55,24 @@
...
+- `master` branch Version string updates
+   - Skip this set of steps if you are patching a previos minor release. 
    - [ ] bump the version(s) in `build/version.go` to `vX.Y.(Z+1)-dev`. 
       - Ensure to update the appropriate version string based on whether you are creating a node release (`NodeBuildVersion`), a miner release (`MinerBuildVersion`), or both.
    - [ ] Run `make gen && make docsgen-cli` before committing changes.
-   - [ ] Remove the contents in the `Unreleased Changelog` section.
-   - [ ] Create a PR with title `build: update Lotus Node|Miner version to vX.Y.Z+1-dev in master`
+   - [ ] Update the CHANGELOG
+     - [ ] Change the `UNRELEASED` section to `UNRELEASED vX.Y.Z` and set the section's content to be `See https://github.com/filecoin-project/lotus/blob/release/vX.Y.Z/CHANGELOG.md`
+     - [ ] Add a new `UNRELEASED` header to top.
```
Comment (BigLep) [inline documentation/misc/RELEASE_ISSUE_TEMPLATE.md:67]: @rjan: I made these changes so there is a pointer to where the UNRELEASED items are (i.e., there in a release branch now)

File: documentation/misc/RELEASE_ISSUE_TEMPLATE.md [hunk 2]
```diff
@@ -57,21 +55,24 @@
    - Link to PR:
    - There likely aren't any changes at this point, but this can be opened with a small whitespace change so the PR is open and we can more easily hold the standard of making improvements incrementally since improvements are usually better done by collecting changes/notes along the way rather than just thinking about it at the end.
    - This will get merged in a `Post Release` step.
-- [ ] (patch release) Fork a new branch (`release/vX.Y.Z` or `release/miner/vX.Y.Z`) from the last stable `release/vX.Y.Z` or `release/miner/vX.Y.Z` and make any further release-related changes to this branch.
+- [ ] (patch release) Fork a new branch (`release/vX.Y.Z` or `release/miner/vX.Y.Z`) from the last stable `release/vX.Y.x` or `release/miner/vX.Y.x` and make any further release-related changes to this branch.
 - [ ] (minor release) Fork a new branch (`release/vX.Y.Z` or `release/miner/vX.Y.Z`) from `master` and make any further release-related changes to this branch.
-- (minor release) `master` branch Version string updates
+- `master` branch Version string updates
+   - Skip this set of steps if you are patching a previos minor release.
```
Comment (BigLep) [inline documentation/misc/RELEASE_ISSUE_TEMPLATE.md:61]: I believe we still need to do these Version strings for path and minor release, unless it's a patch for a previous minor version.

**Review by BigLep — COMMENTED** [review]
@rjan90 : I think this is ready for review.  I left a couple of extra comments.

I really want to get some minimal tooling that creates the release issue based on these questions:
1. Is this for a network upgrade?  If so, what network version?
2. Is this a minor release?
3. Is this a patch release?  If yes, what previous release are you patching?
4. Is this an emergency patch?  (If yes, we can skip RC steps)

With that info, I think we can create a release issue for each of these scenarios:
1. Network upgrade / minor release
2. non-network upgrade / minor release
3. normal patch release against previous minor release
4. normal patch release against current minor release
5. emergency patch release against previous minor release
6. emergency patch release against current minor release

I've got a backlog of these type of improvements, so I don't want to commit to it yet, but I think tooling to even start creating the right set of release steps is a good beachhead towards building "lotus releaser" tooling.

File: documentation/misc/RELEASE_ISSUE_TEMPLATE.md (hunk 2 above)
Comment (rjan90) [reply documentation/misc/RELEASE_ISSUE_TEMPLATE.md:61]: Yes, that is correct. It also reminds me that we should bump the version in master now

**Review by rjan90 — APPROVED** [review]

**Comment by BigLep** [comment]
Merging.  

Followup issue created on tooling to help with release issue creation: https://github.com/filecoin-project/lotus/issues/12477

**Pull request merged**